lib/inline-imports.js

```diff
@@ -74,21 +78,34 @@ function* linesWithInlinedImportsOf(
  * @name inlineImportsWithLineToImports
  * @returns {inlineImports: string, lineToImports: Map<number, {filename: string, line: string}>}
  */
-function inlineImportsWithLineToImports(fileContents, options = {}, visited = new Set()) {
+function inlineImportsWithLineToImports(
+  fileContents,
+  options = {},
+  visited = new Set(),
+  root = true,
+  fragmentCollector = new Map(),
+) {
   const inlineImportsResult = [];
   const lineToImports = new Map();
-
-  for (let { line, match, lineNumber, filename } of linesWithInlinedImportsOf(
+  /** Ideally lineToImports should look like this:
+  // Map<lineNumber, Map<filename,source>>
+**/
+  for (let { line, match, lineNumber, filename, rawSource } of linesWithInlinedImportsOf(
     fileContents,
     options,
     visited,
+    fragmentCollector,
   )) {
     inlineImportsResult.push(line);
 
     // We're only interested in the inlined import lines, ignore any
     // non-matching lines
     if (match) {
-      lineToImports.set(lineNumber, { filename, line });
+      fragmentCollector.set(filename, rawSource);
+      if (root) {
+        lineToImports.set(lineNumber, new Map([...fragmentCollector]));
```

Comment:
FWIW, since `fragmentCollector` is a `Map` you can pass it directly to `new Map` (instead of making a few temporary arrays):

```suggestion
        lineToImports.set(lineNumber, new Map(fragmentCollector));
```